packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -70,26 +74,63 @@ export const Combobox = ({ options, onChange, value, ...restProps }: ComboboxPro
       }
     },
   });
+
+  // the order of middleware is important!
+  const middleware = [
+    flip({
+      // see https://floating-ui.com/docs/flip#combining-with-shift
+      crossAxis: false,
+      boundary: document.body,
+      fallbackPlacements: ['top'],
+    }),
+  ];
+  const elements = { reference: inputRef.current, floating: floatingRef.current };
+  const { floatingStyles } = useFloating({
+    open: isOpen,
+    placement: 'bottom',
+    middleware,
+    elements,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const hasMinHeight = isOpen && rowVirtualizer.getTotalSize() >= MIN_WIDTH;
+
   return (
     <div>
-      <Input suffix={<Icon name={isOpen ? 'search' : 'angle-down'} />} {...restProps} {...getInputProps()} />
-      <div className={styles.dropdown} {...getMenuProps({ ref: listRef })}>
+      <Input
+        suffix={<Icon name={isOpen ? 'search' : 'angle-down'} />}
+        {...restProps}
+        {...getInputProps({ ref: inputRef, onChange: () => {} })}
```

Comment:
You probably need to call the actual onChange provided by Downshift. I played around and got this. 
![image](https://github.com/user-attachments/assets/9ccf9728-5f42-4d1a-ad74-b0fac2bd148a)

1. Select an option
2. Press escapce

Reply:
There is an issue about this in the Downshift repo https://github.com/downshift-js/downshift/issues/718